packages/hardhat-core/src/internal/hardhat-network/provider/input.ts

```diff
@@ -28,6 +28,25 @@ const isRpcDataString = (u: unknown) =>
 const isRpcHashString = (u: unknown) =>
   typeof u === "string" && u.length === 66 && isRpcDataString(u);
 
+interface BlockTagObject {
+  blockHash?: string | Buffer;
+  blockNumber?: string | number | BN;
+}
+
+const isBlockTagObject = (u: any): u is BlockTagObject => {
+  if (typeof u === "object") {
+    if (u?.blockHash != null && u?.blockNumber == null) {
```

Comment:
@alcuadrado should we have a tslint rule that checks that `===` and `!==` are used?

Reply:
@fvictorio this is `==` on purpose to check both `undefined` and `null`, because `null === undefined // false` but `null == undefined // true`

Reply:
Oh, we don't usually do that. If both things can happen, we explicitly check both.

Reply:
For nullish value assertions I'd argue that `==` is the preferred and generally accepted operator in JavaScript / TypeScript and also what powers the nullish coalescing (`.?`) operator (undefined & null are treated equally here too).

For any other scalar comparisons I'd agree.

Reply:
That makes sense, but I think internal consistency is more important in this case. If you make a search, you'll see that there are only a couple of cases where `==` and `!=` are used, and those were probably something that went unnoticed in a code review. I'm surprised we don't have a linter rule for this :sweat_smile: